test/explorer/series/duration_test.exs

```diff
@@ -669,5 +669,37 @@ defmodule Explorer.Series.DurationTest do
              >\
              """
     end
+
+    test "mutate/2 with duration + date" do
+      require Explorer.DataFrame
+      alias Explorer.DataFrame, as: DF
+
+      df =
+        DF.new(aug_20: [~D[2023-08-20]], aug_21: [~D[2023-08-21]])
+        |> DF.mutate(sub: aug_21 - aug_20)
+
+      # Arg-swapping works.
+      df1 = DF.mutate(df, add1: sub + aug_20)
+      assert df1["add1"].dtype == :date
+      assert Series.to_list(df1["add1"]) == [~D[2023-08-21]]
+
+      # Arg-swapping works.
+      df2 = DF.mutate(df, add2: sub + ^df["aug_20"])
+      assert df2["add2"].dtype == :date
+      assert Series.to_list(df2["add2"]) == [~D[2023-08-21]]
+
+      # Arg-swapping works.
+      df2 = DF.mutate(df, add2: sub + aug_20 + sub)
+      assert df2["add2"].dtype == :date
+      assert Series.to_list(df2["add2"]) == [~D[2023-08-22]]
+
+      # Arg-swapping does NOT work.
+      # [src/dataframe.rs:507] &mutations = [
+      #   [(col("sub")) + ([(col("aug_20")) + (col("sub"))])].alias("add2"),
+      # ]
+      df2 = DF.mutate(df, add2: sub + (aug_20 + sub))
+      assert df2["add2"].dtype == :date
+      assert Series.to_list(df2["add2"]) == [~D[2023-08-22]]
+    end
```

Comment:
Hey so I'm a little stuck at this point.

My workaround for `Explorer.Series.add(duration, date)` was to swap the arguments. I tried that same thing with expressions, but now I don't think that will work. There's no way to "see through" the parentheses in expressions like:

```rust
[(col("sub")) + ([(col("aug_20")) + (col("sub"))])]
```

meaning I miss situations where I ought to swap.

Perhaps there's some way to salvage this approach? Or perhaps there's another way to achieve what I'm going for? Any advice would be appreciated!

P.S., here's a pseudo-call-stack when calling `DF.mutate/2`:

<details>

<summary>Click to show/hide</summary>

```elixir
Explorer.DataFrame.mutate(df, mutations) # macro
  Explorer.DataFrame.mutate_with(unquote(df), Explorer.Query.query(unquote(mutations)))
    Explorer.Shared.apply_impl(df, :mutate_with, [df_out, column_pairs])
      Explorer.PolarsBackend.DataFrame.mutate_with(df, out_df, column_pairs)
        Explorer.PolarsBackend.Shared.apply_dataframe(df, out_df, :df_mutate_with_exprs, [exprs, df.groups])
          Explorer.PolarsBackend.Native.df_mutate_with_exprs(df, exprs, groups)
            native/explorer/src/dataframe.rs::df_mutate_with_exprs
```

</details>

I found it helpful when trying to figure out a good place to intervene.